### PR TITLE
Sql editor enhance

### PR DESCRIPTION
**Pull request by JulianWielga**
Process  with sql-editor  should display colorfull sql editor  Mac OS  #0
Process  with sql-editor  should display colorfull sql editor  Mac OS  #1

### CODE REVIEW

File: ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable i18next/no-literal-string */
+import React, {useCallback, useMemo} from "react"
+import {useFFlags} from "../../../../../common/FeatureFlagsUtils"
+import AceWrapper, {AceWrapperProps} from "./AceWrapper"
+
+export default function AceWithFeatureFlags(props: Omit<AceWrapperProps, "noWrap" | "showLines">): JSX.Element {
```
Comment (mproch) [inline ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx:6]: Are these feature flags in our current meaning? we use "feature flags" to be able to hide/show some features from user. Here they can enable them?

Comment (JulianWielga) [reply ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx:6]: maybe not. switches are hidden but accessible under hotkey

Comment (JulianWielga) [reply ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx:6]: it was part of my discussion with @lciolecki. we don't know what is the best so why not give user choice?

File: ui/client/components/graph/node-modal/editors/expression/ExpressionSuggest.js [hunk 2]
```diff
@@ -23,10 +24,23 @@ var inputExprIdCounter = 0
...
+    let token = iterator.getCurrentToken()
+    while (token && (token.type !== "spel.open" && token.type !== "spel.close")) {
+      token = iterator.stepBackward()
+    }
+    return token?.type === "spel.open"
+  }
+  return true
+}
+
+function isSpelTokensAllowed(iterator, modeId) {
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply ui/client/components/graph/node-modal/editors/expression/ExpressionSuggest.js:38]: why complicate? only string is not allowed.

File: ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx (hunk 1 above)
Comment (lciolecki) [reply ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx:6]: In future we can create special page with all feature flags and switches. But right now it's enough.

File: ui/client/actions/nk/featureFlags.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {FeatureFlags} from "../../reducers/featureFlags"
+
+type ToggleFlagsAction = {type: "TOGGLE_FLAGS", keys: Array<keyof FeatureFlags>}
+type SetFlagsAction = {type: "SET_FLAGS", flags: FeatureFlags}
+
+export function toggleFlags(keys: Array<keyof FeatureFlags>): ToggleFlagsAction {
+  return {
+    type: "TOGGLE_FLAGS",
```
Comment (lciolecki) [inline ui/client/actions/nk/featureFlags.ts:8]: I know it's not time.. But I think this type should by enum?

File: ui/client/common/userSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import {useDispatch, useSelector} from "react-redux"
+import {toggleSettings} from "../actions/nk/userSettings"
+import {RootState} from "../reducers"
+import {UserSettings} from "../reducers/userSettings"
+
+export const useUserSettings: () => [UserSettings, (keys: Array<keyof UserSettings>) => void] = () => {
+  const dispatch = useDispatch()
+  const current = useSelector<RootState, UserSettings>(state => state.userSettings)
+  const toggleFeatureFlags = (keys: Array<keyof UserSettings>) => { dispatch(toggleSettings(keys)) }
```
Comment (lciolecki) [inline ui/client/common/userSettings.tsx:9]: `toggleUserFlags` ?

File: ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable i18next/no-literal-string */
+import React, {useMemo} from "react"
+import {useUserSettings} from "../../../../../common/userSettings"
+import AceWrapper, {AceWrapperProps} from "./AceWrapper"
+
+export default function AceWithFeatureFlags(props: Omit<AceWrapperProps, "noWrap" | "showLines">): JSX.Element {
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx:6]: `AceWithCommands`?

**Review by lciolecki — COMMENTED** [review]
@JulianWielga let's look at my comments. I think we are realy close to finish it.

File: ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx (hunk 1 above)
Comment (mproch) [reply ui/client/components/graph/node-modal/editors/expression/AceWithFeatureFlags.tsx:6]: @JulianWielga thanks for separation, can we just rename this class to AceWithSettings?

File: ui/client/common/featureFlags.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import {setFlags, toggleFlags} from "../actions/nk/featureFlags"
+import {store} from "../bootstrap"
+import {RootState} from "../reducers"
+import {FeatureFlags} from "../reducers/featureFlags"
+
+export const useFeatureFlags: () => [FeatureFlags, (keys: Array<keyof FeatureFlags>) => void] = () => {
+  const dispatch = useDispatch()
+  const current = useSelector<RootState, FeatureFlags>(state => state.featureFlags)
+  const toggleFeatureFlags = (keys: Array<keyof FeatureFlags>) => { dispatch(toggleFlags(keys)) }
+  return [current, toggleFeatureFlags]
```
Comment (mproch) [inline ui/client/common/featureFlags.tsx:11]: let's not expose toggleFeatureFlags action, it should be only via console

**Review by mproch — COMMENTED** [review]
Thanks for splitting settings, I have only two minor remarks

**Review by lciolecki — APPROVED** [review]
LGTM

**Pull request merged**